From 7b6ceabfa4c70717a8aedf27c2b59438f3b64ea3 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 6 Jun 2025 20:50:33 +0200
Subject: [PATCH 1/4] phy: move Airoha PCIe PHY driver to dedicated directory

In preparation for additional PHY driver for different Airoha SoC,
move the current PCIe PHY driver in a dedicated directory.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 MAINTAINERS                                     |  4 ++--
 drivers/phy/Kconfig                             | 11 +----------
 drivers/phy/Makefile                            |  4 ++--
 drivers/phy/airoha/Kconfig                      | 13 +++++++++++++
 drivers/phy/airoha/Makefile                     |  2 ++
 drivers/phy/{ => airoha}/phy-airoha-pcie-regs.h |  0
 drivers/phy/{ => airoha}/phy-airoha-pcie.c      |  0
 7 files changed, 20 insertions(+), 14 deletions(-)
 create mode 100644 drivers/phy/airoha/Kconfig
 create mode 100644 drivers/phy/airoha/Makefile
 rename drivers/phy/{ => airoha}/phy-airoha-pcie-regs.h (100%)
 rename drivers/phy/{ => airoha}/phy-airoha-pcie.c (100%)

# diff --git a/MAINTAINERS b/MAINTAINERS
# index 6dbdf02d6b0c..2f30432916a3 100644
# --- a/MAINTAINERS
# +++ b/MAINTAINERS
# @@ -735,8 +735,8 @@ M:	Lorenzo Bianconi <lorenzo@kernel.org>
#  L:	linux-arm-kernel@lists.infradead.org (moderated for non-subscribers)
#  S:	Maintained
#  F:	Documentation/devicetree/bindings/phy/airoha,en7581-pcie-phy.yaml
# -F:	drivers/phy/phy-airoha-pcie-regs.h
# -F:	drivers/phy/phy-airoha-pcie.c
# +F:	drivers/phy/airoha/phy-airoha-pcie-regs.h
# +F:	drivers/phy/airoha/phy-airoha-pcie.c
 
#  AIROHA SPI SNFI DRIVER
#  M:	Lorenzo Bianconi <lorenzo@kernel.org>
diff --git a/drivers/phy/Kconfig b/drivers/phy/Kconfig
index 8d58efe998ec..7f9f5b786643 100644
--- a/drivers/phy/Kconfig
+++ b/drivers/phy/Kconfig
@@ -72,16 +72,6 @@ config PHY_CAN_TRANSCEIVER
 	  functional modes using gpios and sets the attribute max link
 	  rate, for CAN drivers.
 
-config PHY_AIROHA_PCIE
-	tristate "Airoha PCIe-PHY Driver"
-	depends on ARCH_AIROHA || COMPILE_TEST
-	depends on OF
-	select GENERIC_PHY
-	help
-	  Say Y here to add support for Airoha PCIe PHY driver.
-	  This driver create the basic PHY instance and provides initialize
-	  callback for PCIe GEN3 port.
-
 source "drivers/phy/airoha/Kconfig"
 source "drivers/phy/allwinner/Kconfig"
 source "drivers/phy/amlogic/Kconfig"
diff --git a/drivers/phy/Makefile b/drivers/phy/Makefile
index e281442acc75..0ad0e5309493 100644
--- a/drivers/phy/Makefile
+++ b/drivers/phy/Makefile
@@ -10,7 +10,6 @@ obj-$(CONFIG_PHY_LPC18XX_USB_OTG)	+= phy-lpc18xx-usb-otg.o
 obj-$(CONFIG_PHY_XGENE)			+= phy-xgene.o
 obj-$(CONFIG_PHY_PISTACHIO_USB)		+= phy-pistachio-usb.o
 obj-$(CONFIG_USB_LGM_PHY)		+= phy-lgm-usb.o
-obj-$(CONFIG_PHY_AIROHA_PCIE)		+= phy-airoha-pcie.o
 obj-y					+= airoha/	\
 					   allwinner/	\
 					   amlogic/	\
diff --git a/drivers/phy/airoha/Kconfig b/drivers/phy/airoha/Kconfig
new file mode 100644
index 000000000000..1674134ea664
--- a/drivers/phy/airoha/Kconfig
+++ b/drivers/phy/airoha/Kconfig
@@ -1,6 +1,16 @@
 #
 # Phy drivers for Mediatek devices
 #
+config PHY_AIROHA_PCIE
+	tristate "Airoha PCIe-PHY Driver"
+	depends on ARCH_AIROHA || COMPILE_TEST
+	depends on OF
+	select GENERIC_PHY
+	help
+	  Say Y here to add support for Airoha PCIe PHY driver.
+	  This driver create the basic PHY instance and provides initialize
+	  callback for PCIe GEN3 port.
+
 config PHY_AIROHA_USB
 	tristate "Airoha USB PHY Driver"
 	depends on ARCH_AIROHA || COMPILE_TEST
diff --git a/drivers/phy/airoha/Makefile b/drivers/phy/airoha/Makefile
new file mode 100644
index 000000000000..616dc96302af
--- a/drivers/phy/airoha/Makefile
+++ b/drivers/phy/airoha/Makefile
@@ -0,3 +0,4 @@
 # SPDX-License-Identifier: GPL-2.0
 
+obj-$(CONFIG_PHY_AIROHA_PCIE)		+= phy-airoha-pcie.o
 obj-$(CONFIG_PHY_AIROHA_USB)		+= phy-airoha-usb.o
diff --git a/drivers/phy/phy-airoha-pcie-regs.h b/drivers/phy/airoha/phy-airoha-pcie-regs.h
similarity index 100%
rename from drivers/phy/phy-airoha-pcie-regs.h
rename to drivers/phy/airoha/phy-airoha-pcie-regs.h
diff --git a/drivers/phy/phy-airoha-pcie.c b/drivers/phy/airoha/phy-airoha-pcie.c
similarity index 100%
rename from drivers/phy/phy-airoha-pcie.c
rename to drivers/phy/airoha/phy-airoha-pcie.c
-- 
2.48.1

